### ISSUE TITLE
Unable to update VirtualMachineProfile Extension Public Setting (Scale Set)

### ISSUE THREAD

**Issue by pppchris** [Azure/azure-sdk-for-net#3398]
Fluent API, v1.1.1

```
Repo.VirtualMachineScaleSets.GetById(ID);
Set.Update().UpdateExtension("OctopusDeploy").WithPublicSetting("RandomSetting", "Value").Parent().Apply();
```
Updates are missing in the request to Azure. Adding private setting etc works without any issues.

**Comment by anuchandy** [comment]
@pppchris just to clarify, are you saying you could UPDATE the private settings of this extension but unable to update the public settings?

**Comment by pppchris** [comment]
@anuchandy - yes correct, private work.. public does not.

Fiddler tracers are not showing any public setting changes in the API request.

**Comment by anuchandy** [comment]
@pppchris thank you!. Submitted [fix](https://github.com/anuchandy/azure-sdk-for-net/commit/a8b6bca7e14f520983be0b3d0eda29421d96f792) for this issue.

**Comment by martinsawicki** [comment]
looks like the fix made it into v1.2
